gen/build_deploy/bash.py

```diff
@@ -361,6 +361,48 @@ def validate_custom_check_bins_dir(custom_check_bins_dir):
     fi
 }
 
+function d_type_enabled_if_xfs()
+{
+    # Return 1 if $1 is a directory on XFS volume with ftype ! = 1
+    # otherwise return 0
+    DIRNAME="$1"
+
+    RC=0
+    if [[ ! -d "$DIRNAME" ]]; then
+        return $RC
+    fi
+    read -r filesystem_device filesystem_type <<<"$(df -PT "$DIRNAME" | awk 'END{print $1,$2}')"
+    if [[ "$filesystem_type" == "xfs" ]]; then
+        echo -n -e "Checking if "$DIRNAME" is mounted with \"fytpe=1\": "
+        ftype_value="$(xfs_info $filesystem_device | grep -oE ftype=[0-9])"
+        if [[ "$ftype_value" != "ftype=1" ]]; then
+            RC=1
+        fi
+        print_status $RC "${NORMAL}(${ftype_value})"
+    fi
+    return $RC
+}
+
+# check node storage has d_type (ftype=1) support enabled if using XFS
+function check_xfs_ftype() {
+    RC=0
+
+    # If /var/lib/mesos exists check xfs attributes for it
+    (d_type_enabled_if_xfs /var/lib/mesos ) || RC=1
+
+    # If docker root directory exists check xfs attributes for it
+    docker_root_dir="$(docker info | grep 'Docker Root Dir' | cut -d ':' -f 2  | tr -d '[[:space:]]')"
+    (d_type_enabled_if_xfs "$docker_root_dir" ) || RC=1
+
+    # If either /var/lib/mesos or /var/lib/docker do not exist then check /var/lib
+    if [[ (! -d /var/lib/mesos || ! -d /var/lib/docker) ]]; then
```

Comment:
Should we be using `$docker_root_dir` here instead of assuming `/var/lib/docker`?

Reply:
Yes, thats correct. I was initially hard-coding /var/lib/docker but the realized I should not and I only fixed it in one place and missed this one. Will give this some thought on how to do this with $docker_root_dir without over-engineering it. 

Do you know if I it would be a reasonable assumption to make that docker root dir would exist if docker is running. I know that docker is running because we check for that already and if my assumption is true I do not need to traverse up $docker_root_dir one level at a time to see if that exists to then do df on it

Reply:
Yeah I think that's fair to assume. If `docker info` succeeds then the Docker daemon must be running, and I'd expect it to have created its root dir at startup if it didn't exist.